File: packages/app/expoPlugin/ios/ORWidgetReloadBridge.m
//
//  ORWidgetReloadBridge.m
//  Orbit
//
//  Created by Andy Matuschak on 2024-07-12.
//

#import <Foundation/Foundation.h>
#import <React/RCTBridgeModule.h>

NS_ASSUME_NONNULL_BEGIN

@interface RCT_EXTERN_MODULE(WidgetReloadBridge, NSObject)
RCT_EXTERN_METHOD(reloadTimelines)
@end

NS_ASSUME_NONNULL_END

File: packages/app/expoPlugin/ios/ORWidgetReloadBridge.swift
//
//  ORWidgetReloadBridge.swift
//  Orbit
//
//  Created by Andy Matuschak on 2024-07-12.
//

import Foundation
import WidgetKit

@objc(WidgetReloadBridge) class WidgetReloadBridge: NSObject {
  @objc public func reloadTimelines() {
    WidgetCenter.shared.reloadAllTimelines()
    print("Reloading timelines")
  }
}
